### PR TITLE
Scrubbing of Positions works now

### PR DESCRIPTION
**Pull request by pineapplespatula**
I tied the scrubby to the zoom level, which i think i like. this probably needs some tweaking

### CODE REVIEW

File: src/js/jsx/mixin/Scrubby.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                e = e.touches[0];
+            }
+
+            return {
+                x: e.clientX,
+                y: e.clientY
+            };
+        },
+
+        /**
```
Comment (volfied) [inline src/js/jsx/mixin/Scrubby.js]: These functions are unused and should be taken out.

File: src/js/jsx/shared/Label.jsx [hunk 2]
```diff
@@ -25,10 +25,48 @@ define(function (require, exports, module) {
...
     
     var Label = React.createClass({
-        mixins: [React.addons.PureRenderMixin],
+        mixins: [FluxMixin, ScrubbyMixin],
+        
+        propTypes: {
+            size: React.PropTypes.string,
+            onScrub: React.PropTypes.func,
+            onScrubStart: React.PropTypes.func,
+            scrubby: React.PropTypes.bool
```
Comment (volfied) [inline src/js/jsx/shared/Label.jsx]: We can probably get away with removing scrubby and just checking for existence of onScrub function (onScrubStart/onScrubEnd can be optional)

File: src/js/jsx/mixin/Scrubby.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var React = require("react"),
+        _ = require("lodash");
+
+    var math = require("js/util/math");
+
+    module.exports = {
```
Comment (volfied) [inline src/js/jsx/mixin/Scrubby.js]: There was a REALLY useful comment in ColorPicker that explained how to use this mixin. It stated that the user had to provide their own `_updatePosition` function. 

Can you make a similar comment here on how to use this mixin, what the user of this mixin has to provide with function signatures?

**Comment by volfied** [comment]
This is a good start, but needs some cleanup. Since we're providing onScrubStart and onScrub.. we should also provide onScrubEnd, which returns the overall scrub difference so in cases where we don't want to live scrub, we can use onScrubEnd to submit an action.

Great work!~

File: src/js/jsx/mixin/Scrubby.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+        propTypes: {
+            onChange: React.PropTypes.func,
+            max: React.PropTypes.number
```
Comment (volfied) [inline src/js/jsx/mixin/Scrubby.js]: Not sure if this is possible, but can you look into making the _updatePosition that the owner has to implement a required prop so React complains at us if we decide to use this mixin but not provide that function?

Comment (pineapplespatula) [inline src/js/jsx/mixin/Scrubby.js]: it doesn't look like that is possible

File: src/js/jsx/sections/transform/Position.jsx [hunk 5]
```diff
@@ -45,6 +46,9 @@ define(function (require, exports, module) {
             referencePoint: React.PropTypes.string.isRequired
         },
 
+        xScrubbyValue: null,
```
Comment (iwehrman) [inline src/js/jsx/sections/transform/Position.jsx]: It seems like these should be JSDoc's private and also underscore prefixed?

File: src/js/jsx/mixin/Scrubby.js (hunk 4 above)
Comment (volfied) [inline src/js/jsx/mixin/Scrubby.js]: If that's not possible, during componentDidMount, please check for existence of `_updatePosition` and show a warning/error and skip calling it below so it doesn't throw.

File: src/js/jsx/shared/Label.jsx [hunk 6]
```diff
@@ -25,10 +25,47 @@ define(function (require, exports, module) {
...
+                rect = React.findDOMNode(this).getBoundingClientRect();
+
+            var value;
+            if (this.props.vertical) {
+                value = (rect.bottom - clientY) / rect.height;
+            } else {
+                value = (clientX - rect.left) / rect.width;
+            }
+
+            this.props.onScrub(uiStore.zoomCanvasToWindow(value));
```
Comment (volfied) [inline src/js/jsx/shared/Label.jsx]: Alongside checking for _udpatePosition in Scrubby mixin, you should also check for onScrub's existence here.

Comment (volfied) [inline src/js/jsx/shared/Label.jsx]: In fact, this entire function should bail at the beginning if `this.props.onScrub` is not defined. `getBoundingClientRect` is an expensive operation.

File: src/js/jsx/shared/Label.jsx [hunk 7]
```diff
@@ -25,10 +25,47 @@ define(function (require, exports, module) {
...
+            onScrubStart: React.PropTypes.func
+        },
+
+        /**
+         * inform the parent that scrubbing has started
+         *
+         * @private
+         */
+        _updatePositionBegin: function () {
+            this.props.onScrubStart();
```
Comment (volfied) [inline src/js/jsx/shared/Label.jsx]: Ditto for existence checking.

**Comment by volfied** [comment]
Got couple more comments. 

Also, can you hook up onScrubEnd as well, I see it being useful in some situations.

While testing, I saw that we weren't actually transforming the layer until I let go of the mouse, can you look into it as well?

Back to you @pineapplespatula

**Comment by pineapplespatula** [comment]
I'm at a loss as to why we arent transforming the layer during the scrub action. I'll look into this more tomorrow, but my hunch is a core issue

**Comment by pineapplespatula** [comment]
@jsbache diagnosed the drawing issue. This should be ready for another review

**Comment by iwehrman** [comment]
CI checks are failing. This could be a false positive though. Does it check out for you?

**Comment by iwehrman** [comment]
+1. The point of commit messages is to be useful to future humans, which is why we encourage rebasing commits: the end result of most PRs should a single commit with a message that explains the overall change.

**Comment by pineapplespatula** [comment]
@volfied  what hasn't been addressed?

**Comment by volfied** [comment]
Huh, Github must have served me a cached page, I still saw the old code.

**Pull request merged**